spec/unit/tariff_synchronizer/taric_update_spec.rb

```diff
@@ -43,43 +43,76 @@
     end
   end
 
-  describe '.correct_filename_sequence?' do
-    subject(:taric_update) { described_class }
+  describe '.correct_recent_filename_sequence?' do
+    subject(:update) { described_class }
+
+    context 'when there are updates with an unbroken sequence' do
+      before do
+        create(:taric_update, :missing, example_date: Date.parse('2021-12-04'), sequence_number: 'foo')
+        create(:taric_update, :pending, example_date: Date.parse('2021-12-03'), sequence_number: '202')
+        create(:taric_update, :applied, example_date: Date.parse('2021-12-02'), sequence_number: '201')
+        create(:taric_update, :applied, example_date: Date.parse('2021-12-01'), sequence_number: '200')
+      end
+
+      it { is_expected.to be_correct_recent_filename_sequence }
+    end
 
-    before do
-      create(:taric_update, :applied, example_date: applied_date, sequence_number: applied_sequence_number)
-      create(:taric_update, :pending, example_date: pending_date, sequence_number: pending_sequence_number)
+    context 'when there are updates with a broken sequence' do
+      before do
+        create(:taric_update, :pending, example_date: Date.parse('2021-12-03'), sequence_number: '203')
+        create(:taric_update, :applied, example_date: Date.parse('2021-12-02'), sequence_number: '202')
+        create(:taric_update, :applied, example_date: Date.parse('2021-12-01'), sequence_number: '200')
+      end
+
+      it { is_expected.not_to be_correct_recent_filename_sequence }
     end
 
-    let(:applied_date) { Date.parse('2021-12-01') }
-    let(:applied_sequence_number) { '002' }
+    context 'when there are only applied updates' do
+      before do
+        create(:taric_update, :applied, example_date: Date.parse('2021-12-02'), sequence_number: '202')
+        create(:taric_update, :applied, example_date: Date.parse('2021-12-01'), sequence_number: '200')
+      end
+
+      it { is_expected.to be_correct_recent_filename_sequence }
```

Comment:
this was a bit surprising to me since the sequence of the applied was not correct

Reply:
We could push the check for pending updates to the tariff synchronizer apply method.

Reply:
Actually this is already the case